doc/design/status_port.md

```diff
@@ -0,0 +1,453 @@
+    of the `StatusPortData` (either via `StatusPortData` c'tor or
+    `StatusPort{Writer,Reader}` c'tor)
+
+### Discarded ideas & design alternatives
+
+#### Atomic pointer
+
+The pointer to the currently active chunk could also be stored in an `std::atomic`.
+
+```cpp
+std::atomic<T*> activeChunk{nullptr}
+```
+
+However, it would need the full 64-bit and which is not needed when managing
+just two chunks. Hence an `abaCounter` would need to be stored in a separate
+`std::atomic` variable.
+
+#### Omit `ServiceDescription` and discovery
+
+Methods like `offer()` or `stopOffer()` are not needed to keep things simple.
+However, omitting the `ServiceDescription` completely is not possible as it is
+needed to associate the right `StatusPortData` in shared memory with the correct
+writer or reader object on the stack. The `StatusPort` shall follow the
+service-based design.
+
+#### `atomic<T>::exchange` in `StatusPortReader::take()`
+
+A [CAS](https://en.wikipedia.org/wiki/Compare-and-swap) operation is not possible
+because CAS needs write access and a `StatusPortReader` shall not be allowed to
+write data into shared memory.
+
+#### Differentiate between Writer/Reader between RouDi/User?
+
+To be consistent with the previously available publisher and subscribers one
+could argue to follow the same naming. However, the `StatusPort` might be used to
+communicate between ASIL domains in future and hence become user API. As a result
+the naming shall stay Writer/Reader.
+
+#### Can part of the previous port infrastructure be re-used?
+
+Potential candidates:
+
+1. `BasePort`
+   1. The previously available publisher and subscribers follow a much more complex
+   communication pattern. Hence, most of the data in `BasePort` is irrelevant and
+   `StatusPortData` shall stay separate.
+1. `popo::Sample`
+   1. Due to the fact that data is accessed solely via `callable`'s there is no
+   benefit in using the non-owning `popo::Sample` view.
+
+### Solution
+
+#### Class structure
+
+Depicted below is the class diagram.
+
+![StatusPort class diagram](diagrams/status-port/status-port-overview.svg)
+
+The `StatusPortWriter` is a template where `T` is the transferred data type.
+It has a constructor which takes a pointer to the `StatusPortData`
+object in the shared memory management segment. The pointer is acquired in the
+same manner as previous publishers and subscribers via RouDi's unix domain
+socket. The only difference is that one `StatusPortData` object is shared between
+`StatusPortReader` and `StatusPortWriter`. `store()` takes a `function_ref` to
+be able to manipulate the type in-place in shared memory.
+`StatusPortReader` is a template as well and has the same constructor. The first
+and single argument of`take()` is a `function_ref`, which is executed as long as
+the `take()` was unsuccessful. Furthermore, the `StatusPortReader` has the methods
+necessary to be attachable to a `popo::Listener` (e.g. `{enable,disable}Event`).
```

Comment:
You forgot `invalidateTrigger`

Reply:
You overlooked `e.g.` ;) This is not meant to be a complete list.